docs/components/modal.md

```diff
@@ -1,348 +1,752 @@
 
-Tells the component whether or not to render the open modal.
+<KButton @click="modal3Visible = true">Modal</KButton>
+<KModal
+  title="Long modal title gets truncated with an ellipsis"
+  :visible="modal3Visible"
+  @cancel="closeAllModals"
+  @proceed="closeAllModals"
+/>
 
-### title
+```html
+<KModal
+  title="Long modal title gets truncated with an ellipsis"
+  :visible="modalVisible"
+  @cancel="handleModalClose"
+  @proceed="handleModalProceed"
+/>
+```
 
-Text displayed in header if not using slot (**Note**: this field is still required for accessibility reasons even if using the slot).
+### actionButtonText
 
-### hideTitle
+Text to be displayed in action button. Defaults to "Submit".
 
-If not using the `header-content` slot, tells the component whether or not to display the title.
+<KButton @click="modal4Visible = true">Modal</KButton>
+<KModal
+  action-button-text="Ok"
+  :visible="modal4Visible"
+  title="Modal"
+  @cancel="closeAllModals"
+  @proceed="closeAllModals"
+/>
 
-### content
+```html
+<KModal
+  action-button-text="Ok"
+  :visible="modalVisible"
+  title="Modal"
+  @cancel="handleModalClose"
+  @proceed="handleModalProceed"
+/>
+```
 
-Text to display in content if not using slot.
+### actionButtonAppearance
 
-<KButton appearance="primary" @click="defaultIsOpen2 = true">Open Modal</KButton>
+Appearance of action button. See [KButton `appearance`](/components/button#appearance) prop for more details. Defaults to `primary`.
 
-<div>
-  <KModal
-    :isVisible="defaultIsOpen2"
-    hide-title
-    title="I hid my title"
-    content="I am customized!"
-    @proceed="defaultIsOpen2 = false"
-    @canceled="defaultIsOpen2 = false"
-  />
-</div>
+<KButton @click="modal5Visible = true">Modal</KButton>
+<KModal
+  action-button-appearance="danger"
+  :visible="modal5Visible"
+  title="Modal"
+  @cancel="closeAllModals"
+  @proceed="closeAllModals"
+/>
 
 ```html
 <KModal
-  :isVisible="isVisible"
-  hide-title
-  title="I hid my title"
-  content="I am customized!"
-  @proceed="isVisible = false"
-  @canceled="isVisible = false"
+  action-button-appearance="danger"
+  :visible="modalVisible"
+  title="Modal"
+  @cancel="handleModalClose"
+  @proceed="handleModalProceed"
 />
 ```
 
-### textAlign
+### actionButtonDisabled
 
-The alignment for the title and content. Supports `left`, `center` (default), or `right`.
+By default the action button is enabled, but you can pass `true` to this prop should you want to change that.
 
-### maxWidth
-
-The `max-width` of the modal. Default `'500px'`
+<KComponent
+  v-slot="{ data }"
+  :data="{ actionEnabled: true }"
+>
+  <div class="vertical-container">
+    <div>
+      <KButton @click="modal6Visible = true">Modal</KButton>
+    </div>
+    <KInputSwitch
+      v-model="data.actionEnabled"
+      label="Action button enabled"
+    />
+  </div>
 
-### actionButtonText
+  <KModal
+    :action-button-disabled="!data.actionEnabled"
```

Comment:
since we're talking about disabling the button, I think it should be disabled by default in the example. Applies to **all** disabled examples in the other props too